spec/linked_list.js

```diff
@@ -0,0 +1,205 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import LinkedList from '../src/linked_list'
+
+chai.use(chaiChange)
+
+describe('Linked List', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(LinkedList).to.be.a('function')
+  })
+
+  context('getHeadNode()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+    myLinkedList.insert('second')
+    myLinkedList.insert('third')
+
+    it('Returns an object', () => {
+      expect(myLinkedList.getHeadNode())
+        .to.be.a.object
+    })
+    it('Returns the first node in the list', () => {
+      expect(myLinkedList.getHeadNode().data)
+      .to.eql('first')
+    })
+  })
+  context('getTailNode()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+    myLinkedList.insert('second')
+    myLinkedList.insert('third')
+
+    it('Returns an object', () => {
+      expect(myLinkedList.getTailNode())
+        .to.be.a.object
+    })
+    it('Returns the first node in the list', () => {
+      expect(myLinkedList.getTailNode().data)
+      .to.eql('third')
+    })
+  })
+  context('contains()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+    myLinkedList.insert('second')
+
+    it('returns true the set contains the element', () => {
+      expect(myLinkedList.contains('first'))
+      .to.be.true
+    })
+    it('returns true if the element is last in the set', () => {
+      expect(myLinkedList.contains('second'))
+      .to.be.true
+    })
+    it('returns false if the set does not contain the element', () => {
+      expect(myLinkedList.contains('third'))
+      .to.be.false
+    })
+  })
+  context('find()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+    myLinkedList.insert('second')
+
+    it('Returns the first node containing the provided data', () => {
+      expect(myLinkedList.find('second').data)
+      .to.eql('second')
+    })
+    it('Returns -1 if provided data is not found', () => {
```

Comment:
Although technically you did not use the "third" node, I think it would make this test more clear if you tested a node you havent used previously( because it could be anything thats not in the list) like you could test to make sure elephant is not in the list and you should get -1.